clientgen/clientgen.gotpl

```diff
@@ -9,16 +9,20 @@ type Client interface {
 	// the proceeding Client method requests with the given string
 	WithQuery(s string) Client
 
-	{{ range $field := .QueryRoot.Fields -}}
-		{{- if $field.IsResolver }}
-			{{- $field.GoFieldName}}{{ $field.ShortResolverDeclaration }}
-		{{- end }}
-	{{ end }}
-	{{ range $field := .MutationRoot.Fields -}}
-		{{- if $field.IsResolver }}
-			{{- $field.GoFieldName}}{{ $field.ShortResolverDeclaration }}
-		{{- end }}
-	{{ end }}
+    {{- if .HasQuery -}}
+        {{ range $field := .QueryRoot.Fields -}}
+            {{- if $field.IsResolver }}
+                {{- $field.GoFieldName}}{{ $field.ShortResolverDeclaration }}
+            {{- end }}
+        {{ end }}
+    {{ end }}
+	{{- if .HasMutation -}}
```

Comment:
Same here, you can just do `if .MutationRoot`

Reply:
I tried that and got..

`error generating go client: clientgen.gotpl: template: clientgen.gotpl:19:8: executing "clientgen.gotpl" at <.MutationRoot>: MutationRoot has arguments but cannot be invoked as function`

Reply:
@mikehepple this is because this template uses `.MutationRoot.Fields` in a couple of places that must also be wrapped with an `if` statement:  Particularly around line 95 of this file

Reply:
@marwan-at-work Just got it working on my schema (by skipping the query generation) and found that, submitted another commit. The error I got was at line 19 (the first reference to `.MutationRoot.Fields`)